### PR TITLE
Add truncate util

### LINKED ISSUE #685: Bug: Buggy repo listing
### Describe the bug
When a repo name is very long, it breaks the whole UI (very easy to reproduce on mobile)

### To Reproduce

1. Use a device or change the window to a small width (mobile is best)
1. Add a repo with a long name
### Expected behavior
The text doesn't overflow.

### Screenshots
screenshot with long name

### Desktop

- OS: Win 10
- Browser: Firefox Developer Edition
- Version: Latest

### Smartphone

- Device: iPhone Xs
- OS: iOS 13.5.1
- Browser: Safari
- Version: latest

### Additional context
Setting a max length wouldn't be the best solution -> I would limit the display size - not the stored size

### PR DESCRIPTION
**Pull request by yg**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [ ] ✨ Feature
- [x] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

Add truncate util with three params

```js
truncate(string, length, ending) // ending is default to (...)
```

#685

## Related Tickets & Documents

Closes #685

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

Screenshot

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by yg** [comment]
@bdougie What can be the default **length** we can use?

**Comment by bdougie** [comment]
> This can be easily fixed by truncating URLs to 100 characters.

**Comment by yg** [comment]
Now with length `100`

Screenshot

File: src/components/RepoListItem.js [hunk 1]
```diff
@@ -12,7 +13,7 @@ function RepoListItem({goal, stars}) {
         <FlexCenter>
           <Avatar alt="avatar" src={`https://avatars.githubusercontent.com/${goal.title.split("/")[0].replace(/\s+/g, "")}`} />
           <FlexColumn className="details">
-            <p>{goal.title.replace(/\s+/g, "")}</p>
+            <p>{truncate(goal.title.replace(/\s+/g, ""), 100)}</p>
```
Comment (bdougie) [inline src/components/RepoListItem.js:16]: I am thinking 100 hundred is too much per your screenshot. Can you bump it down to 60?

File: src/lib/truncate.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export function truncate(str, length, ending) {
+  if (length == null) {
+    length = 100;
+  }
+  if (ending == null) {
+    ending = "...";
+  }
```
Comment (bdougie) [inline src/lib/truncate.js:7]: Could these conditionals also be written like so?

```js
export function truncate(str, length = 100, ending = "...") {
```

**Comment by yg** [comment]
Now with length `60`

Screenshot

**Pull request merged**